### DISCUSSION TITLE
Construct an array from the results a subquery in postgres?

### DISCUSSION THREAD

**Discussion by kyle-mccarthy** [SeaQL/sea-query#588] [Q&A]
Is it possible to create an array from the results of a subquery using the postgres-array feature?

Simple example:

```
SELECT ARRAY (SELECT * FROM (VALUES (1), (2), (3), (4)) AS vals (id)) AS vals_array
```

Returns:

|vals_array|
| --------- |
|`{1,2,3,4}`|

The last paragraph in [this section](https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-ARRAY-CONSTRUCTORS) of the postgres docs discusses it.

**Comment by billy1624** [comment]
Hey @kyle-mccarthy, welcome to SeaQL! Sorry for the delay. Something like this?

```rs
#[derive(Iden)]
#[iden = "ARRAY"]
pub struct ArrayFunc;

let sub_select = Query::select()
    .expr(Expr::asterisk())
    .from(Char::Table)
    .to_owned();

let select = Query::select()
    .expr(Func::cust(ArrayFunc).arg(SimpleExpr::SubQuery(
        None,
        Box::new(sub_select.into_sub_query_statement()),
    )))
    .to_owned();

assert_eq!(
    select.to_string(PostgresQueryBuilder),
    r#"SELECT ARRAY((SELECT * FROM "character"))"#
);
```
